web/sharings/sharings.go

```diff
@@ -111,6 +111,39 @@ func GetSharing(c echo.Context) error {
 	return jsonapi.Data(c, http.StatusOK, as, nil)
 }
 
+// GetSharingsInfoByDocType returns, for a given doctype, all the sharing
+// information, i.e. the involved sharings and the shared documents
+func GetSharingsInfoByDocType(c echo.Context) error {
+	inst := middlewares.GetInstance(c)
+	docType := c.Param("doctype")
+	sharingIDs, err := sharing.GetSharingsByDocType(inst, docType)
+	if err != nil {
+		return wrapErrors(err)
+	}
+	res := make([]*sharing.APISharing, len(sharingIDs))
+
+	sharings, err := sharing.FindSharings(inst, sharingIDs)
+	if err != nil {
+		return wrapErrors(err)
+	}
+	for i, s := range sharings {
+		if err = checkGetPermissions(c, s); err != nil {
+			return wrapErrors(err)
+		}
+		docs, err := sharing.GetSharedDocsBySharingID(inst, s.ID())
```

Comment:
It doesn't look optimal. Maybe you can modify `GetSharingsByDocType` to build a map of `sharingID -> []SharedDoc`, and just use the map here to fill the SharedDocs field of the APISharing